model/src/main/proto/exploration.proto

```diff
@@ -322,6 +322,46 @@ message UserAnswer {
   WrittenTranslationContext written_translation_context = 6;
 }
 
+// Corresponds to a raw item selection answer that user has selected.
+message ItemSelectionRawAnswer {
+  repeated int32 selected_indexes = 1;
+}
+
+// Corresponds to a raw representation of the current answer entered by the user which is used to
+// retain state on configuration changes.
+message RawUserAnswer {
+  oneof answer_input_type {
+    // A raw answer entered by user in text input interaction.
+    string text = 1;
+    // A raw answer entered by user in fraction input interaction.
+    string fraction = 2;
+    // A raw answer entered by user in numeric input interaction.
+    string numeric = 3;
+    // A raw answer entered by user in ratio input interaction.
+    string ratio_input = 4;
+    // A raw answer entered by user in ratio input interaction.
+    string math_expression = 5;
+    // A user's selection for item selection and multiple choice interactions.
+    ItemSelectionRawAnswer item_selection = 6;
+    // A user's selection for drag and drop interactions.
+    DragAndDropRawAnswer drag_and_drop = 7;
+    // A user's selection for image region selection interaction.
+    ImageWithRegions.LabeledRegion image_region_selection = 8;
+  }
+}
+
+// Corresponds to a current drag and drop answer
+message DragAndDropRawAnswer {
```

Comment:
As per the issues found by @rt4914 I have to move back to the older implementation of storing the list of GroupSubtitledHtml. We can't directly use content id as for DragAndDropSort we not just changing the location of content id but we are alos merging and the logic to merge two content ids require corresponding html text also. So, it is mandatory to use GroupSubtitledHtml

Reply:
Sorry, I don't understand what you mean by this explanation. The interaction itself only tracks content IDs--that should be sufficient to represent exactly the groups and order of elements. Could you maybe clarify why we can't just store/retrieve the exact state of the interaction which is just a list of SetOfTranslatableHtmlContentIds per:

https://github.com/oppia/oppia-android/blob/ac68f842d3f9c1816f0e0c1ff3e5e9b9ea165774/app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/DragAndDropSortInteractionViewModel.kt#L231.

Reply:
Used listofSetOftranslatableIds

Reply:
Resolving this comment now I have applied proposed solution.